r.in.wcs.worker/r.in.wcs.worker.py

```diff
@@ -161,10 +173,58 @@ def main():
     os.remove(tif)
     tif = tif.replace(".0", ".tif")
     RM_FILES.append(tif)
-    try:
-        urlretrieve(url, tif)
-    except URLError:
-        grass.fatal(_(f"Failed to reach the server.\nURL: {url}"))
+
+    num_retry_no_connection = 0
+    while num_retry_no_connection <= num_retry_max:
+        try:
+            num_retry_unstable_connection = 0
+            urlretrieve(url, tif)
+            gdalinfo_err, gdalinfo_returncode = get_gdalinfo_returncodes(tif)
+            if (
+                gdalinfo_returncode != 0
+                or ("TIFFReadEncodedStrip" in gdalinfo_err)
+                or ("TIFFReadEncodedTile" in gdalinfo_err)
+            ):
+                if num_retry_unstable_connection == num_retry_max:
+                    grass.fatal(
+                        _(
+                            "Failed to download tif after "
+                            f"{num_retry_max} retries."
+                        )
+                    )
+                grass.warning(
+                    _(
+                        f"Broken tif downloaded, with error {gdalinfo_err}."
+                        " Try to re-download. Retry "
+                        f"{num_retry_unstable_connection}/{num_retry_max} ..."
+                    )
+                )
+                sleep(5)
+                os.remove(tif)
+                urlretrieve(url, tif)
+                gdalinfo_err, gdalinfo_returncode = get_gdalinfo_returncodes(
+                    tif
+                )
+                num_retry_unstable_connection += 1
```

Comment:
```suggestion
                os.remove(tif)
                num_retry_unstable_connection += 1
```
`urlretrieve` is already called at the beginning of the `try` block, also `get_gdalinfo_returncodes`